weakmap.c

```diff
@@ -761,12 +868,12 @@ wkmap_aset(VALUE self, VALUE key, VALUE val)
  *  deletes the entry, and returns the associated value:
  *    m = ObjectSpace::WeakKeyMap.new
  *    m["foo"] = 2
- *    h.delete("foo") { |key| raise 'Will never happen'} # => 2
+ *    m.delete("foo") { |key| raise 'Will never happen'} # => 2
```

Comment:
This is a bad example (not your fault, it was wrong before your typo fix).  Without frozen string literals, the `"foo"` refer to separate objects, and the code raises.  Probably easiest to use a local variable to ensure a reference to the same object.

Reply:
Note that this is `WeakKeyMap`, it does use regular equality semantic, so it doesn't matter if it's the same instance of `"foo"`.

Reply:
Good point.  However, without frozen string literals, couldn't the `"foo"` string passed to `m.[]` be garbage collected before the `m.delete` call?

Reply:
If the `dupstring` endup up triggering GC, then yes it's possible. But it's like with the `GC.start` examples, it may not work if there is something on the stack than happen to holding the `VALUE`.

Which raise the same question of how accurate code snippets need to be, and also why I'm not a big fan of using code snippet to explain weak references. But again, I don't consider myself good at documenting so I don't have a strong opinion either way.

Reply:
Seems like assigning to a local variable and using that local variable for `m.[]` would be best in this case, since then you can be sure the value would not be garbage collected.  You don't need the variable for `m.delete`, since, as you mentioned, it does not compare by identity.